src/material/slider/slider.ts

```diff
@@ -625,705 +152,757 @@ export class MatSlider
 
-  /** Starts observing and updating the slider if the host changes its size. */
-  private _observeHostResize() {
-    if (typeof ResizeObserver === 'undefined' || !ResizeObserver) {
-      return;
+      console.log('updated min and max');
+
+      eInput._updateWidthInactive();
+      sInput._updateWidthInactive();
+    } else {
+      const eInput = this._getInput(Thumb.END);
+      if (eInput) {
+        eInput._updateThumbUIByValue();
+      }
     }
 
-    // MDC only updates the slider when the window is resized which
-    // doesn't capture changes of the container itself. We use a resize
-    // observer to ensure that the layout is correct (see #24590 and #25286).
-    this._ngZone.runOutsideAngular(() => {
-      this._resizeObserver = new ResizeObserver(entries => {
-        // Triggering a layout while the user is dragging can throw off the alignment.
-        if (this._isActive()) {
-          return;
-        }
+    this._updateTickMarkUI();
+    this._updateTickMarkTrackUI();
+    this._cdr.detectChanges();
+  }
 
-        clearTimeout(this._resizeTimer);
-        this._resizeTimer = setTimeout(() => {
-          // The `layout` call is going to call `getBoundingClientRect` to update the dimensions
-          // of the host. Since the `ResizeObserver` already calculated them, we can save some
-          // work by returning them instead of having to check the DOM again.
-          if (!this._isActive()) {
-            this._cachedHostRect = entries[0]?.contentRect;
-            this._layout();
-            this._cachedHostRect = null;
-          }
-        }, 50);
-      });
-      this._resizeObserver.observe(this._elementRef.nativeElement);
-    });
+  /** Whether or not the slider thumbs overlap. */
+  private _thumbsOverlap: boolean = false;
+
+  /** Returns true if the slider knobs are overlapping one another. */
+  private _areThumbsOverlapping(): boolean {
+    const startInput = this._getInput(Thumb.START);
+    const endInput = this._getInput(Thumb.END);
+    if (!startInput || !endInput) {
+      return false;
+    }
+    return endInput.translateX - startInput.translateX < 20;
   }
 
-  /** Whether any of the thumbs are currently active. */
-  private _isActive(): boolean {
-    return this._getThumb(Thumb.START)._isActive || this._getThumb(Thumb.END)._isActive;
+  /**
+   * Updates the class names of overlapping slider thumbs so
+   * that the current active thumb is styled to be on "top".
+   */
+  private _updateOverlappingThumbClassNames(source: MatSliderRangeThumbInterface): void {
+    const sibling = source.getSibling()!;
+    const sourceThumb = this._getThumb(source.thumbPosition);
+    const siblingThumb = this._getThumb(sibling.thumbPosition);
+    siblingThumb._hostElement.classList.remove('mdc-slider__thumb--top');
+    this._thumbsOverlap
+      ? sourceThumb._hostElement.classList.add('mdc-slider__thumb--top')
+      : sourceThumb._hostElement.classList.remove('mdc-slider__thumb--top');
```

Comment:
You should be able to simplify this to `sourceThumb._hostElement.classList.toggle('mdc-slider__thumb--top', this._thumbsOverlap);`.